components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -177,7 +178,10 @@ class AppLinksUseCases(
 
             val appIntent = when (intent.data) {
                 null -> null
-                else -> intent
+                else -> when (alwaysDeniedSchemes.contains(intent?.data?.scheme)) {
+                    true -> null
+                    else -> intent
+                }
```

Comment:
nit: it may be worth flattening this into a single when, e.g. something like

```Kotlin
            val appIntent = when {
                intent.data == null -> null
                alwaysDeniedSchemes.contains(intent?.data?.scheme) -> null
                else -> intent
          }
```

Reply:
Agreed.  I'll flatten it.